pkg/vault/vault.go

```diff
@@ -22,6 +23,57 @@ type Client interface {
 	GetSecretVersionFromVault(secret string) (int64, error)
 }
 
+// GetVaultToken uses a service account to get a vault auth token
+func (a *gcpAuthClient) GetVaultToken(vc *vaultClient) (string, error) {
+	if vc.config.traceEnabled {
+		var span *trace.Span
+		vc.ctx, span = trace.StartSpan(vc.ctx, fmt.Sprintf("%s/getVaultToken", vc.config.tracePrefix))
+		defer span.End()
+	}
+
+	var err error
+
+	if vc.config.githubAuth {
+		vaultResp, err := githubVaultAuth(vc)
+		if err != nil {
+			return "", err
+		}
+
+		log.Debug("Successfully logged into Vault with auth/github/login")
+
+		return vaultResp.Auth.ClientToken, nil
+	}
+
+	if vc.config.googleAuth {
+		a.iamService, err = iam.NewService(vc.ctx)
+		if err != nil {
+			log.Debugf("initialze client: getting new iam service: %v", err)
+			return "", fmt.Errorf("getting new iam service: google: could not find default credentials")
+		}
+
+		log.Debug("Successfully created IAM client")
+
+		err = a.generateSignedJWT(vc.ctx, vc.config)
+		if err != nil {
+			log.Debugf("generating signed jwt: %v", err)
+			return "", fmt.Errorf("generating signed jwt, sigining jwt: Post")
+		}
+
+		log.Debug("Successfully generated signed JWT")
+
+		vaultResp, err := a.gcpSaAuth(vc)
+		if err != nil {
+			return "", err
+		}
+
+		log.Debug("Successfully logged into Vault with auth/gcp/login")
+
+		return vaultResp.Auth.ClientToken, nil
+	}
+
+	return "ERROR", err
```

Comment:
returning an empty string alongside an actual error is preferred here. you can do that like with something like this:

```
return "", errors.New("GetVaultToken: configuration error, one of [githubAuth, googleAuth] must be set to true")
```

Reply:
Good call, added and tests are passing!